cohorts/load.py

```diff
@@ -397,14 +412,16 @@ def is_lambda(func):
             for key, value in on.iteritems():
                 col, df = apply_func(on=value, col=key, df=df)
                 cols.append(col)
-            return (cols, df)
         if type(on) == list:
             cols = []
             for i, elem in enumerate(on):
                 col = elem.__name__ if not is_lambda(elem) else "column_%d" % i
                 col, df = apply_func(on=elem, col=col, df=df)
                 cols.append(col)
-            return (cols, df)
+
+        if (rename_cols):
+            df = df.rename(columns=_strip_column_names(df.columns, keep_paren_contents=keep_paren_contents))
```

Comment:
Thanks - was wondering what to do about `cols` -- this will be a bit tricky to do cleanly since I won't know at this point if the column-rename was successful... feel like I should look these up in the dictionary returned by the original _strip_column_names on the df. 

I'll try to write up a test case for this.